packages/fluence-js/src/internal/compilerSupport/v3impl/callFunction.ts

```diff
@@ -105,15 +116,18 @@ const isReturnTypeVoid = (def: FunctionCallDef) => {
  */
 const extractArgs = (
     args: any[],
-    numberOfExpectedArgs: number,
+    def: FunctionCallDef,
 ): {
     peer: FluencePeer;
     config?: FnConfig;
-    args: any[];
+    args: { [key: string]: any };
 } => {
+    const argumentTypes = getArgumentTypes(def);
+    const numberOfExpectedArgs = Object.entries(argumentTypes).length;
```

Comment:
I would suggest having a separate variable named "argumentTypesEntries" so there is no need to do Object.entries twice

Reply:
it's a cheap operation but still I see this =)

Reply:
Also you don't ever use "entries". Just use Object.keys instead